### ISSUE TITLE
[ENH] add logic to restrict passing negative fh if `"capability:insample": False` is set

### ISSUE THREAD

**Issue by yarnabrina** [sktime/sktime#5961]
According to the registry, `"capability:insample"` tag indicates `"can the forecaster make in-sample predictions?"`.

Currently even if this tag is set, `test_predict_time_index_in_sample_full` test still continues to run and all such tests fail. There are multiple forecasters in `tests/_config.py` specifically disabling this test to avoid CI failure.

I suggest we add a logic in testing so that these tests are not triggered in the first place, and use `_config` only to list tests where we actually do not know what is going wrong or fixing is difficult.

**Comment by fkiraly** [comment]
Yes, I think the contract here is not documented - the test should pass if the forecaster raises a `NotImplementedError` upon receiving an in-sample `fh` - the conditoin should be `not fh.is_all_out_of_sample()`.

This should probably be done by the boilerplate layer, instead of being part of the extension contract not communicated clearly to the extender developer.

I would *still* test it, namel that the correct exception is raised. I.e., if the tag is `True`, test correct return type; if `False`, check the correct exception.

**Comment by fkiraly** [comment]
hm, that would raise an error at read. I think it should already be raised at write. Since, don't we already know the first time we see `fh` whether it is in-sample?

**Comment by sahusiddharth** [comment]
Hello @fkiraly, how about using a setter, that can handle it while writing?

**Comment by sahusiddharth** [comment]
@fkiraly, What will happen if a list is passed, what must be done it has some negative numbers, all negative numbers?

**Comment by fkiraly** [comment]
instead of doing that manually, I would use the method `not fh.is_all_out_of_sample(self.cutoff)`,
